custom_components/tahoma/climate.py

```diff
@@ -67,179 +179,63 @@
-            preset_temp = {
-                PRESET_FREEZE: None,
-                PRESET_AWAY: None,
-                PRESET_ECO: None,
-                PRESET_COMFORT: None,
-            }
-            if DEVICE_CLASS_TEMPERATURE in options:
-                if device.url in options[DEVICE_CLASS_TEMPERATURE]:
-                    sensor_id = options[DEVICE_CLASS_TEMPERATURE][device.url]
-                    if device.widget == ST:
-                        entities.append(
-                            TahomaClimate(device, controller, sensor_id, preset_temp)
-                        )
-                    else:
-                        for k, v in preset_temp.items():
-                            if (
-                                k in options[DEVICE_CLASS_TEMPERATURE]
-                                and options[DEVICE_CLASS_TEMPERATURE][k] != ""
-                            ):
-                                preset_temp[k] = float(
-                                    options[DEVICE_CLASS_TEMPERATURE][k]
-                                )
-                        entities.append(
-                            TahomaClimate(device, controller, sensor_id, preset_temp)
-                        )
-
-            else:
-                entities.append(
-                    TahomaClimate(device, controller, preset_temp=preset_temp)
-                )
-
-    async_add_entities(entities)
 
 
-async def update_listener(hass, entry):
-    """Handle options update."""
-    options = dict(entry.options)
-    for entity in hass.data["climate"].entities:
-        if entity.unique_id in options[DEVICE_CLASS_TEMPERATURE]:
-            entity.set_temperature_sensor(
-                options[DEVICE_CLASS_TEMPERATURE][entity.unique_id]
-            )
-            if entity.widget != ST:
-                preset_temp = {
-                    PRESET_FREEZE: None,
-                    PRESET_AWAY: None,
-                    PRESET_ECO: None,
-                    PRESET_COMFORT: None,
-                }
-                for k, v in preset_temp.items():
-                    if (
-                        k in options[DEVICE_CLASS_TEMPERATURE]
-                        and options[DEVICE_CLASS_TEMPERATURE][k] != ""
-                    ):
-                        preset_temp[k] = float(options[DEVICE_CLASS_TEMPERATURE][k])
-                entity.set_preset_temperatures(preset_temp)
-            entity.schedule_update_ha_state()
-
-
-class TahomaClimate(TahomaDevice, ClimateEntity):
-    """Representation of a Tahoma thermostat."""
-
-    def __init__(self, tahoma_device, controller, sensor_id=None, preset_temp=None):
-        """Initialize the sensor."""
-        super().__init__(tahoma_device, controller)
-        if preset_temp is None:
-            preset_temp = {}
-        self._temp_sensor_entity_id = sensor_id
-        self._current_temperature = 0
-        self._target_temp = None
+class SomfyThermostat(BaseClimateEntity):
```

Comment:
Can you move these new classes to a dedicated file? It will avoid having a big module.

Reply:
to be honest, I tried but it failed miserably... Somehow I couldn't get HA to start and correctly import the classes...

Reply:
done